src/rocks_compaction_scheduler.cpp

```diff
@@ -75,26 +153,34 @@ namespace mongo {
         virtual std::string name() const override { return _name; }
         virtual void run() override;
 
+        void compact(const CompactOp& op);
+
         rocksdb::DB* _db;  // not owned
+        RocksCompactionScheduler* _compactionScheduler;  // not owned
 
         bool _compactionThreadRunning = true;
         stdx::mutex _compactionMutex;
         stdx::condition_variable _compactionWakeUp;
-        std::deque<CompactOp> _compactionQueue;
+        using CompactQueue =
+            std::priority_queue<CompactOp, std::vector<CompactOp>, std::greater<CompactOp>>;
+        CompactQueue _compactionQueue;
     };
 
     const char* const CompactionBackgroundJob::_name = "RocksCompactionThread";
 
-    CompactionBackgroundJob::CompactionBackgroundJob(rocksdb::DB* db)
-        : _db(db) {
+    CompactionBackgroundJob::CompactionBackgroundJob(rocksdb::DB* db,
+                                                     RocksCompactionScheduler* compactionScheduler)
+        : _db(db), _compactionScheduler(compactionScheduler) {
         go();
     }
 
     CompactionBackgroundJob::~CompactionBackgroundJob() {
         {
             stdx::lock_guard<stdx::mutex> lk(_compactionMutex);
             _compactionThreadRunning = false;
-            _compactionQueue.clear();
+            // Clean up the queue
+            CompactQueue tmp;
+            _compactionQueue.swap(tmp);
```

Comment:
Interesting, is that faster than `clear`?

Reply:
It's just because `std::priority_queue` doesn't have `clear()` method.